packages/core-data/src/types/menu-location.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { Context, WithoutNevers } from './common';
+
+interface FullMenuLocation< C extends Context > {
+	/**
+	 * The name of the menu location.
+	 */
+	name: string;
+	/**
+	 * The description of the menu location.
+	 */
+	description: string;
+	/**
+	 * The ID of the assigned menu.
+	 */
+	menu: number;
+}
+
+export type MenuLocation< C extends Context > = WithoutNevers<
+	FullMenuLocation< C >
+>;
```

Comment:
if we know we don't have any `never` values we can skip the wrapping again.

```ts
export type MenuLocation<C extends Context> = FullMenuLocation<C>;
```

the argument could also be made here that we don't need the wrapper at all and instead of creating the `FullMenuLocation` we only create `MenuLocation`